src/tools/wasm-merge.cpp

```diff
@@ -477,13 +481,20 @@ struct InputMergeable : public ExpressionStackWalker<InputMergeable, Visitor<Inp
         }
       }
     }
+    // Copy over the non-imports
     for (auto& curr : wasm.functions) {
-      curr->type = ftNames[curr->type];
-      assert(curr->type.is());
-      outputMergeable.wasm.addFunction(curr.release());
+      if (curr) {
```

Comment:
Added a comment - at this point we have transferred the imports, and left nullptrs in their place. So anything left must be a defined function.

We can't use iterDefinedFunctions since it would segfault checking if a nullptr is imported. It might be cleaner to just clean up those nullptrs I suppose. I think it's ok with a comment though?

Reply:
Ah, yeah makes sense. And the assert as well suggests "we should not have any imports here", so sgtm